alg/teca_temporal_reduction.py

```diff
@@ -254,24 +254,96 @@ def update(self, out_array, in_array):
                         if out_array.itemsize < 8 else \
                         out_array.astype(float64)
 
-                # accumulate
-                return out_array + in_array
+                # identify the invalid values
+                if self.fill_value is not None:
+                    out_is_bad = np.isclose(out_array, self.fill_value)
+                    in_is_bad = np.isclose(in_array, self.fill_value)
+
+                # initialize the count the first time through. this needs to
+                # happen now since before this we don't know where invalid
+                # values are.
+                if self.count is None:
+                    if self.fill_value is None:
+                        self.count = 1.0
+                    else:
+                        self.count = np.where(out_is_bad, np.float32(0.0),
+                                              np.float32(1.0))
+
+                if self.fill_value is not None:
+                    # update the count only where there is valid data
+                    self.count += np.where(in_is_bad, np.float32(0.0),
+                                           np.float32(1.0))
+
+                    # accumulate
+                    tmp = np.where(out_is_bad, np.float32(0.0), out_array) \
+                        + np.where(in_is_bad, np.float32(0.0), in_array)
+
+                else:
+                    # update count
+                    self.count += np.float32(1.0)
+
+                    # accumulate
+                    tmp = out_array + in_array
+
+                return tmp
 
             def finalize(self, out_array):
-                n = self.count
-                self.count = 1.0
-                return out_array / n
+                if self.fill_value is not None:
+                    # finish the average. We keep track of the invalid
+                    # values (these will have a zero count) set them to
+                    # the fill value
+                    n = self.count
+                    ii = np.isclose(n, np.float32(0.0))
+                    n[ii] = np.float(1.0)
```

Comment:
I think that numpy's typical default is that `np.float` refers to `np.float64`, so I think this mixes 64-bit floats in the 32-bit float array.  I'm not actually sure how/whether python handles type coercion in this case; it may not matter.

Reply:
done